rfcs/rfc-shared-components.md

```diff
@@ -0,0 +1,196 @@
+# RFC For BCRS Shared Components
+
+- Start Date: (2020-08-07)
+- Target Major Version: None
+- Reference Issues: None
+- Entity Issue: https://github.com/bcgov/entity/issues/4551
+- Implementation PR: None
+
+# Summary
+
+Using current best practices and available tools/packages, create a library of _shared_ Vue (UI) components that can be
+used by different UI projects.
+
+In particular, this library/framework/architecture will contain (simple) business-focused components (eg, BaseAddress,
+list of directors, entity info, etc) that won't be in every app but will get reused. This is in contrast to the
+https://github.com/bcgov/sbc-common-components library that contains components that will probably be in every BCRS app
+and which contains more complex components that use services, etc. Ultimately, there may be a thin line between the
+proposed and existing library and, in the future, they could possibly be merged.
+
+Another goal is to be able to individually version each component. This would allow components to be individually
+updated without forcing the consuming project to import all other changes to the library. This should make smaller
+rollouts easier.
+
+Whether the proposed approach is a good one -- or needs tweaking -- should become clear when we start to use it and
+accumulate numerous (increasingingly complex) components over time.
+
+# Prototype
+
+BCRS Shared Components is a multi-package Lerna repository of shared Vue components, each published individually and
+that you can explore/ develop/ document/ test using Storybook.
+
+The prototype repository is here: https://github.com/bcgov/bcrs-shared-components
+
+The Storybook pages for this project is here: https://bcgov.github.io/bcrs-shared-components/
+
+# Motivation
+
+1) Reuse existing functionality in new projects
+
+2) Reduce duplicated code / fix bugs in one place only
+
+3) Common design / styles between apps
+
+4) Simpler component development than in sbc-common-components
+
+5) Easier component maintainability than in sbc-common-components
+
+6) Independent versioning of each component
+
+# Main Options Considered
+
+1) Duplicated code between projects
+
+    Don't actually do this if possible!
+
+2) Component inheritance / mixins / plugins / slots (ie, within Vue framework itself)
+
+    We are already doing some of this but it's a compromise between development effort and value, and is within
+    individual projects only.
+
+3) Git submodules
+
+    Is not generally recommended.
+
+4) Multiple repositories
+
+    Lots of overhead.
+
+5) Monorepo
+
+    All projects/components/packages are within a single repository.
+
+    Sbc-common-components is a library example of this (using npm alone). However, additional tools can facilitate
+    management of multiple projects, packages, libraries, components - see below.
+
+# Additional Tools Considered
+
+1) https://github.com/features/packages
```

Comment:
pretty sure that javascript packages is actually npm under the covers, as they acquired it last March or something like that.